rpc_state_reader_sn_api/src/lib.rs

```diff
@@ -0,0 +1,994 @@
+        });
+
+        let resp: RpcResponseFelt = self.rpc_call(&params).unwrap();
+
+        resp.result
+    }
+
+    fn get_storage_at(&self, contract_address: &ContractAddress, key: &StorageKey) -> StarkFelt {
+        let contract_address = contract_address.0.key();
+        let key = key.0.key();
+        let params = ureq::json!({
+            "jsonrpc": "2.0",
+            "method": "starknet_getStorageAt",
+            "params": [contract_address.to_string(),
+            key.to_string(), self.block.to_value()],
+            "id": 1
+        });
+
+        let resp: RpcResponseFelt = self.rpc_call(&params).unwrap();
+
+        resp.result
+    }
+
+    /// Requests the given transaction to the Feeder Gateway API.
+    pub fn get_transaction_receipt(&self, hash: &TransactionHash) -> RpcTransactionReceipt {
+        let params = ureq::json!({
+            "jsonrpc": "2.0",
+            "method": "starknet_getTransactionReceipt",
+            "params": [hash.to_string()],
+            "id": 1
+        });
+        self.rpc_call::<RpcResult<RpcTransactionReceipt>>(&params)
+            .unwrap()
+            .result
+    }
+}
+
+mod utils {
+    use std::io::{self, Read};
+
+    use cairo_lang_utils::bigint::BigUintAsHex;
+    use starknet::core::types::{LegacyContractEntryPoint, LegacyEntryPointsByType};
+    use starknet_api::deprecated_contract_class::{EntryPoint, EntryPointType};
+
+    use super::*;
+
+    #[derive(Debug, Deserialize)]
+    pub struct MiddleSierraContractClass {
+        pub sierra_program: Vec<BigUintAsHex>,
+        pub contract_class_version: String,
+        pub entry_points_by_type: ContractEntryPoints,
+    }
+
+    pub(crate) fn map_entry_points_by_type_legacy(
+        entry_points_by_type: LegacyEntryPointsByType,
+    ) -> HashMap<EntryPointType, Vec<EntryPoint>> {
+        let entry_types_to_points = HashMap::from([
+            (
+                EntryPointType::Constructor,
+                entry_points_by_type.constructor,
+            ),
+            (EntryPointType::External, entry_points_by_type.external),
+            (EntryPointType::L1Handler, entry_points_by_type.l1_handler),
+        ]);
+
+        let to_contract_entry_point = |entrypoint: &LegacyContractEntryPoint| -> EntryPoint {
+            let felt: StarkFelt = StarkHash::new(entrypoint.selector.to_bytes_be()).unwrap();
+            EntryPoint {
+                offset: EntryPointOffset(entrypoint.offset as usize),
+                selector: EntryPointSelector(felt),
+            }
+        };
+
+        let mut entry_points_by_type_map = HashMap::new();
+        for (entry_point_type, entry_points) in entry_types_to_points.into_iter() {
+            let values = entry_points
+                .iter()
+                .map(to_contract_entry_point)
+                .collect::<Vec<_>>();
+            entry_points_by_type_map.insert(entry_point_type, values);
+        }
+
+        entry_points_by_type_map
+    }
+
+    use flate2::bufread;
```

Comment:
Why is this here? If this isn't intentional maybe the autoimport feature got confused and place it here instead of the top of the module.
If it is intentional I assume it's used only for the function below. In that case it should be placed within the function's body since that way it won't pollute the entire module's namespace.